### DISCUSSION TITLE
Make `output: hybrid` the default and remove `output: static`

### DISCUSSION THREAD

**Discussion by Princesseuh** [withastro/roadmap#999] [Proposal]
### Body

# Summary

Make the behaviour of `output: hybrid` the default, removing the static options. The option (which could be renamed) would instead only control the default prerendering state per page / endpoint

# Background & Motivation

Due to output: `hybrid` being added later than `static` and `server`, it always felt a bit wonky to have all three options:

- static: fully prerendered
- hybrid: prerendered by default, can opt out per page
- server: fully server rendered, can opt out page

Apart from technicalities, there's no actual reason for having static and hybrid be separate, as (again, apart from technicalities) result in the same output.

What I'd suggest is that the config end up looking like this instead:

```diff
{
--  output: 'static' | 'hybrid' | 'server' = 'static'
++  prerenderDefault: true | false = true,
}
```

An adapter would still be required to use actual SSR features in production, and you'd get warned / it'd block the build appropriately if you're missing one. (probably a warning in dev, a full block in build)

# Goals

- Reduce the difference between SSG and SSR, thus reducing the friction, especially in development, between switching between the modes
- Promote the usage of SSR sparingly, which goes in tune with Server Islands, Actions etc.
- Make it easier to document SSR in general, right now there's a lot of "In SSR, if your page is prerendered..." that could potentially be reduced to just "If your page is prerendered"

**Comment by matthewp** [comment]
I think we should do this, but leave the option as `output: 'static' | 'server'` for now. While I do like `prerenderDefault` a lot, I think it's a step too far for most people right now who are not used to thinking in these terms. Once we have hybrid as the default we can revisit the name of the option in the future.

**Comment by targetlucked69** [comment]
Will you still need an adapter for a purely prerendered `hybrid` app? Because currently, both `server` and `hybrid` needs an adapter, and currently a `static` output dont.

**Reply by ematipico** [reply]
You would need an adapter only if you have on-demand routes, AKA routes that aren't prerendered

**Comment by jasikpark** [comment]
One plus for `'static'` is that you can ensure you have a fully pre-rendered site that doesn't rely on any SSR capabilities.

As in, the build fails if you try to enable SSR routes.

 I guess a fully-static `'hybrid'` doesn't require an adapter?

**Reply by Princesseuh** [reply]
You still need to opt-in to SSR if you don't have `output: 'server'`, if you try to use a SSR feature on a SSG route, you'll get an error

**Reply by jasikpark** [reply]
As in this you mean?

```js
export const prerender = false;
```

**Reply by Princesseuh** [reply]
Yes. If you make a page server-rendered without having an adapter, Astro will warn / error telling you that an adapter is required for a server build
